src/Update.php

```diff
@@ -154,16 +162,45 @@ public function getCurrents()
         return $currents;
     }
 
+    /**
+     * Verify the database schema integrity by running the related CLI command
+     * @return bool True if the schema was validated and OK, false otherwise
+     */
+    private function checkSchemaIntegrity(): bool
+    {
+        // Initialize console application
+        $console = new \Glpi\Console\Application();
+        $output = new Symfony\Component\Console\Output\BufferedOutput();
+        $input = new \Symfony\Component\Console\Input\ArrayInput([
+            '--check-all-migrations' => true
+        ]);
+        try {
+            $this->migration->displayMessage(__('Checking database schema integrity...'), 'strong');
+            $result = $console->find('glpi:database:check_schema_integrity')->run($input, $output);
+            if ($result !== 0) {
+                $this->migration->displayError(__('The database schema is not consistent with the current GLPI version.'));
+                $this->migration->displayMessage($output->fetch(), 'strong');
+            }
+        } catch (\Exception $e) {
+            $this->migration->displayError(__('Unable to check database schema integrity.'));
+            $this->migration->displayError($e->getMessage());
+            return false;
+        }
+
+        return $result === 0;
+    }
```

Comment:
Instead of running the command, maybe adding `hasDifferences` and `getDatabaseDiff` methods that would accept a `string $schema_filepath` parameter would be better. It would permit to add tests on this part, and to call directly the checker here, instead of calling a command.